src/app/metadata/services/massQuery.service.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Metadata search service
+ * @copyright The GHGA Authors
+ * @license Apache-2.0
+ */
+
+import { HttpClient } from '@angular/common/http';
+import { Injectable, Signal, computed, inject, resource, signal } from '@angular/core';
+import { ConfigService } from '@app/shared/services/config.service';
+import { firstValueFrom } from 'rxjs';
+import { SearchResults, emptySearchResults } from '../models/search-results';
+
+/**
+ * Metadata search service
+ *
+ * This service provides the functionality to fetch search results from the server via MASS.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class MassQueryService {
```

Comment:
I suggest using simpler names for the services.

Think of separation of concerns:

Only the services themselves need to known that there are "metldata" and "mass" backend APIs and how they need to be used. The users of the service (the feature components) don't need to know anything about the backend API.

The names of the services also don't need to be too specific unless it's needed to distinguish them.

That's why I suggested to name the services "metadataService" and "metadataSearchService". In principle they could also be merged into one service. But search is a bit more complex and distinct feature, so you can also argue that they should be kept separate.

Reply:
Done 👍